express-zod-api/src/zod-plugin.ts

```diff
@@ -93,46 +107,74 @@ const objectMapper = function (
           R.map(([key, value]) => R.pair(tool[String(key)] || key, value)),
           R.fromPairs,
         );
-  const nextShape = transformer(R.clone(this.shape)); // immutable
-  const output = z.object(nextShape)[this._def.unknownKeys](); // proxies unknown keys when set to "passthrough"
+  const nextShape = transformer(R.clone(this._zod.def.shape)); // immutable
+  const hasPassThrough = this._zod.def.catchall instanceof z.ZodUnknown;
+  const output = (hasPassThrough ? z.looseObject : z.object)(nextShape); // proxies unknown keys when set to "passthrough"
+  // @ts-expect-error -- ignoring inconsistency of Extra type
   return this.transform(transformer).pipe(output);
 };
 
 if (!(metaSymbol in globalThis)) {
   (globalThis as Record<symbol, unknown>)[metaSymbol] = true;
-  Object.defineProperties(z.ZodType.prototype, {
-    ["example" satisfies keyof z.ZodType]: {
-      get(): z.ZodType["example"] {
-        return exampleSetter.bind(this);
+  for (const entry of Object.keys(z)) {
+    if (!entry.startsWith("Zod")) continue;
```

Comment:
some of them might not extend ZodType: ZodError, ZodFunction. Check again. Maybe could ensure that or make a list of exceptions.

Reply:
addressed in 02167a98